### DISCUSSION TITLE
`Buffer is not defined` error when connecting WalletConnect

### DISCUSSION THREAD

**Discussion by puffdood** [wevm/wagmi#163] [Q&A] [answered]
Does anyone encounter `Buffer is not defined` error when connecting WalletConnect and possibly how to fix this? The Wallet Connect pop-up never appeared and it returns this error instead. I am using React with typescript.

**Comment by tmm** [comment, marked as answer]
What framework (Next.js, Remix, etc.) and/or bundler (Vite, esbuild, etc.) are you using?

WalletConnect (and WalletLink) relies on polyfills of native Node globals ("process") and modules ("buffer", "events") which might not be automatically included by your setup.

**Reply by puffdood** [reply]
Hi @tmm  thanks for the reply. I am using create react-app with typescript.

I am not too familiar with web frameworks. So, you're saying there are missing dependencies I need to install? I tried to add buffer but it does not seem to help.

**Reply by rcstanciu** [reply]
@puffdood After installing, you must polyfill `Buffer` as it is not done automatically. For example, you can do it with the following snippet
```javascript
global.Buffer = global.Buffer || require("buffer").Buffer
```

before importing WalletConnect (somewhere at the root; `index.js`/`index.tsx` should be the safest place).

**Reply by puffdood** [reply]
@rcstanciu This works. Thank you. Really appreciate the help.

**Comment by SarahKay99** [comment]
I have the same issue but only when I haven't downloaded Coinbase Wallet into my browser extensions. When I have both Coinbase Wallet and Metamask in my extensions, everything works fine. Is there a solution which allows WalletConnect to work regardless of what other extensions are downloaded?

**Reply by tmm** [reply]
See https://github.com/wagmi-dev/wagmi/discussions/163#discussioncomment-2177209. My guess is the Coinbase Wallet extension adds that to the window when it's installed.

**Reply by SarahKay99** [reply]
I fixed it by adding the Buffer code. Very weird how this all works.

```javascript
import { Buffer } from "buffer";

if (!window.Buffer) {
    window.Buffer = Buffer;
}
```
